email/sender.go

```diff
@@ -97,9 +99,13 @@ func NewSender(ctx context.Context, db *database.DB, logger *logrus.Logger, deps
 	if err != nil {
 		return Sender{}, errors.AddContext(err, "failed to parse email config")
 	}
+	sess, err := db.NewSession()
+	if err != nil {
+		return Sender{}, err
+	}
 	return Sender{
 		staticConfig: c,
-		staticCtx:    ctx,
+		staticCtx:    mongo.NewSessionContext(ctx, sess),
```

Comment:
This is needed, so when the sender is counting how many emails it needs to send, it doesn't need to create an ad-hoc session. Creating an ad-hoc session causes a potential data race with mongo's `Client().NumberSessionsInProgress()` which can't cause any damage but it breaks our tests.